Challenge-4/Bot_1/src/agent.ts

```diff
@@ -15,60 +16,87 @@ import {
+  return async function initialize() {
+    const network = await provider.getNetwork();
+    configuratorProxy = await getConfigurator(network.chainId);
+    networkManager = new NetworkManager(data);
+    await networkManager.init(provider);
+  };
+}
 
 export function provideHandleGovernanceTransaction(
   provider: ethers.providers.Provider,
   assetAbi: string[],
 ): HandleTransaction {
   return async function HandleTransaction(tx: TransactionEvent) {
     const finding: Finding[] = [];
-    const network = await provider.getNetwork();
-    let configuratorProxy = await getConfigurator(network.chainId);
-    const eventFilterMapping: {
-      [key: string]: ReturnType<typeof tx.filterLog>;
-    } = {
-      BORROW_KINK: tx.filterLog(BORROW_KINK, configuratorProxy),
-      SUPPLY_KINK: tx.filterLog(SUPPLY_KINK, configuratorProxy),
-      SET_GOVERNOR: tx.filterLog(SET_GOVERNOR, configuratorProxy),
-      BORROW_PYIR: tx.filterLog(BORROW_PYIR, configuratorProxy),
-      SUPPLY_PYIR: tx.filterLog(SUPPLY_PYIR, configuratorProxy),
-      BORROW_CF: tx.filterLog(BORROW_CF, configuratorProxy),
-      LIQUIDATE_CF: tx.filterLog(LIQUIDATE_CF, configuratorProxy),
-    };
-    const assetToken = await getAddress(provider);
-    const changedEvents: { [key: string]: any } = {};
 
-    // Fetch old value and new value of the changed event. For some events, that index value differs, hence the logic
-    Object.entries(eventFilterMapping).forEach(([key, value]) => {
-      if (Array.isArray(value) && value.length > 0) {
-        if (key != "LIQUIDATE_CF" && key != "BORROW_CF") {
-          changedEvents[key] = {
-            Old_value: value[0].args[1].toString(),
-            New_value: value[0].args[2].toString(),
-          };
-        } else {
-          changedEvents[key] = {
-            Old_value: value[0].args[2].toString(),
-            New_value: value[0].args[3].toString(),
-          };
-        }
+    const result = tx.filterLog(
+      [
+        BORROW_KINK,
+        SUPPLY_KINK,
+        SET_GOVERNOR,
+        BORROW_PYIR,
+        SUPPLY_PYIR,
+        BORROW_CF,
+        LIQUIDATE_CF,
+      ],
+      configuratorProxy,
+    );
+
+    const assetToken = await getAddress(networkManager);
+    const changedEvents: { [key: string]: any } = {};
+    result.forEach((log) => {
+      const name = log.name;
+      if (
+        name != "UpdateAssetBorrowCollateralFactor" &&
+        name != "UpdateAssetLiquidateCollateralFactor"
+      ) {
+        changedEvents[name] = {
+          Old_value: log.args[1].toString(),
+          New_value: log.args[2].toString(),
+        };
+      } else {
+        changedEvents[name] = {
+          Old_value: log.args[2].toString(),
+          New_value: log.args[3].toString(),
+        };
       }
     });
 
     // Additional feature - Fetch collateral asset for the most transacted token in the pool i.e. USDC
-    const obj = await getCollateralAsset(
+    const infoObject = await getCollateralAsset(
```

Comment:
```suggestion
    const collateralAssets = await getCollateralAsset(
```
Suggestion, as the function is named `getCollateralAssets` (per my suggestion), the variable will be more congruent.